diff --git a/src/core.c b/src/core.c
index b3010ce4..93b915aa 100644
--- a/src/core.c
+++ b/src/core.c
@@ -5233,4 +5233,15 @@ static void *GamepadThread(void *arg)

     return NULL;
 }
+
+void SetKeyCallback(void(*f)(int, int, int, int))
+{
+#if defined(PLATFORM_DESKTOP) || defined(PLATFORM_WEB)
+    glfwSetKeyCallback(CORE.Window.handle, f);
+#else  # RPI
+    // Not sure how to handle this yet :-(
+#endif
+}
+
+
 #endif      // PLATFORM_RPI
